Revise the `Getting Started Monitoring` guide to support Scalar Manager (#248)
scalar-labs/helm-charts: docs/getting-started-monitoring.md

## Step 2. Prepare a custom values file

> 1. Get the sample file [scalar-prometheus-custom-values.yaml](./conf/scalar-prometheus-custom-values.yaml) for `kube-prometheus-stack`.

Reviewer: ```suggestion
1. Save the sample file [scalar-prometheus-custom-values.yaml](./conf/scalar-prometheus-custom-values.yaml) for `kube-prometheus-stack`.
```

   * Example
     ```yaml
     alertmanager:

       service:
         type: LoadBalancer

     ...

     grafana:

       service:
         type: LoadBalancer
         port: 3000

     ...

     prometheus:

       service:
         type: LoadBalancer

     ...
     ```
   * Note:

>        * If you want to customize the Prometheus Operator deployment using Helm Charts, you need to set the following configuration for monitoring Scalar products.
>            * The `serviceMonitorSelectorNilUsesHelmValues` and `ruleSelectorNilUsesHelmValues` must be set to `false` (`true` by default) to make Prometheus Operator detects `ServiceMonitor` and `PrometheusRule` of Scalar products.
> 

Reviewer: As I was making changes to the second second-level bullet, I realized this bullet needed similar changes for consistency.

```suggestion
       * If you want to customize the Prometheus Operator deployment by using Helm Charts, you'll need to set the following configurations to monitor Scalar products:
           * Set `serviceMonitorSelectorNilUsesHelmValues` and `ruleSelectorNilUsesHelmValues` to `false` (`true` by default) so that Prometheus Operator can detect `ServiceMonitor` and `PrometheusRule` in Scalar products.
```

Reviewer 2: Sorry, I'm not familiar with the correct grammar of English, but I have one question.

Strictly, the `ServiceMonitor` and `PrometheusRule` are NOT components of Scalar products. In other words, Scalar products do NOT include these components in themselves. They are separated resources and created by Scalar Helm Chart as Kubernetes resources as follows.

```
+---[Kubernetes]-----------------------------------------------------------------------------------------------+
|                                                                                                              |
| +------------------------+                                                                                   |
| | ScalarDB Cluster (Pod) |                                                                                   |
| +------------------------+                                                                                   |
|                                                                                                              |
| +------------------------+                                                                                   |
| | ScalarDL Ledger  (Pod) |                                                                                   |
| +------------------------+                                                                                   |
|                                                                                                              |
| +------------------------+                                                                                   |
| | ScalarDL Auditor (Pod) |                                                                                   |
| +------------------------+                                                                                   |
|                                                                                                              |
| +---------------------------------------+                                                                    |
| | ServiceMonitor (for ScalarDB Cluster) | <--- Created by Scalar Helm Chart for ScalarDB Cluster monitoring. |
| +---------------------------------------+      But this ServiceMonitor is not component of Scalar products.  |
|                                                It is one of the Kubernetes resources.                        |
| +---------------------------------------+                                                                    |
| | PrometheusRule (for ScalarDB Cluster) | <--- Created by Scalar Helm Chart for ScalarDB Cluster monitoring. |
| +---------------------------------------+      But this PrometheusRule is not component of Scalar products.  |
|                                                It is one of the Kubernetes resources.                        |
|                                                                                                              |
| +---------------------------------------+                                                                    |
| | ServiceMonitor (for ScalarDL Ledger)  | <--- Created by Scalar Helm Chart for ScalarDL Ledger monitoring.  |
| +---------------------------------------+      But this ServiceMonitor is not component of Scalar products.  |
|                                                It is one of the Kubernetes resources.                        |
| +---------------------------------------+                                                                    |
| | PrometheusRule (for ScalarDL Ledger)  | <--- Created by Scalar Helm Chart for ScalarDL Ledger monitoring.  |
| +---------------------------------------+      But this PrometheusRule is not component of Scalar products.  |
|                                                It is one of the Kubernetes resources.                        |
|                                                                                                              |
| +---------------------------------------+                                                                    |
| | ServiceMonitor (for ScalarDL Auditor) | <--- Created by Scalar Helm Chart for ScalarDL Auditor monitoring. |
| +---------------------------------------+      But this ServiceMonitor is not component of Scalar products.  |
|                                                It is one of the Kubernetes resources.                        |
| +---------------------------------------+                                                                    |
| | PrometheusRule (for ScalarDL Auditor) | <--- Created by Scalar Helm Chart for ScalarDL Auditor monitoring. |
| +---------------------------------------+      But this PrometheusRule is not component of Scalar products.  |
|                                                It is one of the Kubernetes resources.                        |
|                                                                                                              |
+--------------------------------------------------------------------------------------------------------------+
```

In this case, I feel it would be better to use `of` or `for` instead of `in` to avoid confusions.

```diff
- Prometheus Operator can detect `ServiceMonitor` and `PrometheusRule` in Scalar products.
+ Prometheus Operator can detect `ServiceMonitor` and `PrometheusRule` of Scalar products.
```

```diff
- Prometheus Operator can detect `ServiceMonitor` and `PrometheusRule` in Scalar products.
+ Prometheus Operator can detect `ServiceMonitor` and `PrometheusRule` for Scalar products.
```

What do you think?
(If I miss some English grammar things, please correct me! 🙇‍♀️)

Reviewer: @kota2and3kan I see. Thank you for the explanation! 

With that in mind, I think using `for` would be accurate:

```suggestion
       * If you want to customize the Prometheus Operator deployment by using Helm Charts, you'll need to set the following configurations to monitor Scalar products:
           * Set `serviceMonitorSelectorNilUsesHelmValues` and `ruleSelectorNilUsesHelmValues` to `false` (`true` by default) so that Prometheus Operator can detect `ServiceMonitor` and `PrometheusRule` for Scalar products.

```

Reviewer 2: @josh-wong 
I see. Looks good!
Thank you for the update!

>        * If you use Scalar Manager, you need to set the following configurations to support Scalar Manager to collect CPU and memory resources.
>            * The `kubeStateMetrics.enabled`, `nodeExporter.enabled`, and `kubelet.enabled` must be set to `true`.

Reviewer: I'd like to clarify this bullet, but I'm not sure which is accurate. I've added a couple suggestions below.

Suggestion 1:

```suggestion
       * If you want to use Scalar Manager, you'll need to set the following configurations so that Scalar Manager can collect CPU and memory resources:
           * Set `kubeStateMetrics.enabled`, `nodeExporter.enabled`, and `kubelet.enabled` to `true`.
```

Suggestion 2:

```suggestion
       * If you want to use Scalar Manager, you'll need to set the following configurations to enable Scalar Manager to collect CPU and memory resources:
           * Set `kubeStateMetrics.enabled`, `nodeExporter.enabled`, and `kubelet.enabled` to `true`.
```

Author: @josh-wong 

Thank you!

Both of them are accurate for me.
I would take the second one because `enable` looks more technical w

Reviewer: @supl Oops, I originally had another suggestion that meant something else but I forgot to change the sentence `I'd like to clarify this bullet, but I'm not sure which is accurate.` 

Both of the suggestions above mean the same thing, so either is OK👍

Reviewer 2: Both of them look good to me, and I also feel the second one is better too!
